skimage/segmentation/_join.py

```diff
@@ -33,15 +39,30 @@ def join_segmentations(s1, s2):
     array([[0, 1, 3, 2],
            [0, 5, 3, 2],
            [4, 5, 5, 3]])
+    >>> join_segmentations(s1, s2, return_mapping=True)
+    (array([[0, 1, 3, 2],
+           [0, 5, 3, 2],
+           [4, 5, 5, 3]]), ArrayMap(array([0, 1, 2, 3, 4, 5]), array([0, 0, 1, 1, 2, 2])), ArrayMap(array([0, 1, 2, 3, 4, 5]), array([0, 1, 0, 1, 0, 1])))
```

Comment:
This is hard to read for someone who doesn't already know about `ArrayMap`, so I might instead suggest:

```python
>>> j, m1, m2 = join_segmentations(s1, s2, return_mapping=True)
>>> m1
ArrayMap(array([0, 1, 2, 3, 4, 5]), array([0, 0, 1, 1, 2, 2]))
>>> np.all(m1[j] == s1)
True
>>> np.all(m2[j] == s2)
True
```

What do you think?

Reply:
> But if you don't overwrite j, you can move it to after the return clause and all is well, right?
Yes, I renamed the variables (hopefully not too verbose) and moved the line after the return.

Reply:
> What do you think?

That example is much better, I updated it.